### PR TITLE
Protective MBR support (#1)

### LINKED ISSUE #858: TODO: testcase for protective MBR filesystems
In case a `mkfs` command (udf, vfat) creates a new protective MBR or let's say a partition table in general that the caller wasn't originally expecting and which overlaps with the filesystem boundaries, and the caller then operates on that new partition, rewriting its data and consequently destroying the partition table, ... bad things will happen.

Write a testcase for cross-filesystem types multiple format operations and ensure that all objects are consistent and sane.
AFAIK for the moment we only have separate test cases for each filesystem type, wiping the whole disk between testcases.

### PR DESCRIPTION
**Pull request by tbzatek**
This is a first attempt to fix corner cases where protective partition table is created by `mkfs` tools. See #859 for details.

Fixes #858

### CODE REVIEW

**Comment by tbzatek** [comment]
TODO: might be a good time to break the `libblockdev` 3.x API and change the `bd_fs_wipe()` arguments. Then rebase this pull request.

**Comment by tbzatek** [comment]
I've dropped the controversial `BLKRRPART` ioctl fatal errors as it was unreliable on various kernel versions. The idea behind it was to easily detect whether the block device was busy. 

We're now at the previous behaviour: after wiping the block device that should remove any partitions on it, the `BLKRRPART` ioctl would fail as a warning and a `Error synchronizing after initial wipe: Timed out waiting for object` error will be reported as the partitions would still be present. No real difference other than the bogus message and slight delay due to waiting.

File: src/tests/dbus-tests/test_80_filesystem.py [hunk 1]
```diff
@@ -731,7 +741,90 @@ def test_userspace_mount_options(self):
...
+        if not self._can_create:
+            self.skipTest('Cannot create %s filesystem' % self._fs_name)
 
+        disk = self.get_object('/block_devices/' + os.path.basename(self.vdevs[0]))
+        self.assertIsNotNone(disk)
+        self.addCleanup(self.wipe_fs, self.vdevs[0])
+        if self._fs_name == "vfat":
+            # need to force creation of a protective MBR
+            ret, _out = self.run_command('mkfs.vfat --mbr=y %s' % self.vdevs[0])
+            self.assertEqual(ret, 0)
```
Comment (vojtechtrefny) [inline src/tests/dbus-tests/test_80_filesystem.py:758]: We might need a sleep or udev settle here, looks like the tests are failing quite consistently here and I assume it's probably caused by a slow reaction to the external `mkfs` call.

**Comment by tbzatek** [comment]
`util-linux` 2.37 changes that will need to be reflected in the tests: https://github.com/karelzak/util-linux/commit/751eca28fcfa1e325770904d7181b22e48c72b29

File: src/tests/dbus-tests/test_80_filesystem.py (hunk 1 above)
Comment (tbzatek) [reply src/tests/dbus-tests/test_80_filesystem.py:758]: Good point, added 2 seconds.

**Comment by tbzatek** [comment]
I think I'm done with the changes here, ready for review. The test failures about already mounted filesystem are unrelated and likely a problem of mounting externally while `udisksd` not registering the change in time.

**Comment by vojtechtrefny** [comment]
> TODO: might be a good time to break the `libblockdev` 3.x API and change the `bd_fs_wipe()` arguments. Then rebase this pull request.

The libblockdev API change is now merged.

**Review by vojtechtrefny — APPROVED** [review]
Looks good to me.

**Comment by tbzatek** [comment]
Rebased for the recent `bd_fs_wipe()` API changes. See https://github.com/storaged-project/libblockdev/pull/648 for just another related issue.
